Libraries/Opc.Ua.Gds.Server.Common/ApplicationsNodeManager.cs

```diff
@@ -580,6 +592,49 @@ out nextRecordId
             return ServiceResult.Good;
         }
 
+        private ServiceResult OnRevokeCertificate(
+             ISystemContext context,
+             MethodState method,
+             NodeId objectId,
+             NodeId applicationId,
+             byte[] certificate)
+        {
+            HasApplicationAdminAccess(context);
+
+            if (m_database.GetApplication(applicationId) == null)
+            {
+                return new ServiceResult(StatusCodes.BadNotFound, "The ApplicationId does not refer to a registered application.");
+            }
+            if (certificate == null || certificate.Length == 0 )
+            {
+                throw new ServiceResultException(StatusCodes.BadInvalidArgument, "The certificate is not a Certificate for the specified Application that was issued by the CertificateManager.");
+            }
+
```

Comment:
The spec mentions Bad_UserAccessDenied check be part of the RevokeCertificate(). Should this be checked?

Reply:
Line 602 HasApplicationAdminAccess(context); does this check